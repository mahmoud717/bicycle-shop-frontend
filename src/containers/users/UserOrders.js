/* eslint-disable no-nested-ternary */
/* eslint-disable react/jsx-key */
/* eslint-disable react/prop-types */
import React from 'react';

const UserOrders = ({ authData }) => {
  if (authData.loading) {
    return (
      <h1 className="text-center">loading</h1>
    );
  }
  return (
    <div className="container-fluid order-card-container d-flex flex-column justify-content-center align-items-center">
      {authData.loggedIn ? authData.userOrders.length === 0
        ? <div className="no-orders">You don&apos;t have any orders, Make A purchase to make it appear here. </div>
        : authData.userOrders.map(order => (
          <div key={order.id} className="order-card d-flex">
            <div className="order-card-image">
              <img src={order.product_image_url} alt="" className="w-100" />
            </div>
            <div className="order-card-details">
              <div className="order-card-name">{order.product_name}</div>
              <div className="order-card-data">{order.created_at}</div>
            </div>

          </div>

        )) : ''}
    </div>
  );
};

export default UserOrders;
