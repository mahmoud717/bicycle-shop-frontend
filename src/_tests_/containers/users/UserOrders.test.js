import React from 'react';

import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import UserOrders from '../../../containers/users/UserOrders';

const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <UserOrders />
  </Provider>,
);

describe('UserOrders Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the UserOrders container', () => {
    const page = component.find('.order-card-container');
    expect(page).toMatchSnapshot();
  });
});
