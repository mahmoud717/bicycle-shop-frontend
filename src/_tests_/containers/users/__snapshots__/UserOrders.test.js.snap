// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserOrders Component renders the UserOrders container 1`] = `ShallowWrapper {}`;
